src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVDUSTFilteredKmerizer.java

```diff
@@ -0,0 +1,82 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.stream.Stream;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.SVKmer.Base;
+
+/** An iterator over kmers with a specified maximum DUST-style, low-complexity score.
+ *
+ *  DUST scores a sequence by counting the number of occurrences of each of the 64 possible trimers in that sequence,
+ *  and summing these counts according to:  sumOverCounts( count[t]*(count[t]-1)/2 ) / (sequence.length - 3)
+ *
+ *  Since we're dealing with kmers of a fixed sequence length, we ignore the division by sequence length.
+ *  So our DUST-like score is (kSize-3) too big, but proportional to a real DUST score for the kmer,
+ *  which is good enough and much faster to calculate.
+ */
+public class SVDUSTFilteredKmerizer extends SVKmerizer {
+    private final int maxDUSTScore;
+    private final int[] trimerCounts;
+    private int curDUSTScore;
+
+    public SVDUSTFilteredKmerizer(final byte[] seq, final int kSize, final int maxDUSTScore, final SVKmer kmer ) {
+        this(new ASCIICharSequence(seq), kSize, maxDUSTScore, kmer);
+    }
+
+    public SVDUSTFilteredKmerizer(final CharSequence seq, final int kSize, final int maxDUSTScore, final SVKmer kmer ) {
+        super(kSize, seq);
+        if ( kSize < 4 ) {
+            throw new GATKException("kmer size must be at least 4 for this filter to work properly.");
+        }
+        this.maxDUSTScore = maxDUSTScore;
+        trimerCounts = new int[64]; // there are 64 different trimers
+        final int polyACount = kSize - 2; // the current kmer is poly-A, so trimer 0 (AAA) has kSize-2 counts
+        trimerCounts[0] = polyACount;
+        curDUSTScore = polyACount*(polyACount - 1)/2; // see formula above
+        nextKmer = nextKmer(kmer, 0);
+    }
+
+    public static Stream<SVKmer> stream( final CharSequence seq, final int kSize, final int maxDUSTScore, final SVKmer kmer ) {
+        return Utils.stream(new SVDUSTFilteredKmerizer(seq, kSize, maxDUSTScore, kmer));
+    }
+
+    public static Stream<SVKmer> stream( final byte[] seq, final int kSize, final int maxDUSTScore, final SVKmer kmer ) {
+        return Utils.stream(new SVDUSTFilteredKmerizer(seq, kSize, maxDUSTScore, kmer));
+    }
+
+    @Override
+    protected SVKmer nextKmer( SVKmer tmpKmer, int validBaseCount ) {
```

Comment:
not totally related, but [this](https://www.youtube.com/watch?v=ZWRRXP-XFvY) is always fun to watch, enjoy